src/components/settings/settings/SettingsPanel.tsx

```diff
@@ -0,0 +1,33 @@
+import { Box, Button, Divider, Flex, HStack, SimpleGrid, Spacer, Text, useDisclosure } from '@chakra-ui/react';
+import useSubjects from '../../../hooks/useSubjects';
+import { SubjectType } from '../../../entities/Subject';
+import { AddIcon } from '@chakra-ui/icons';
+import { useParams } from 'react-router-dom';
+import { YearCourseDeleteModal } from './YearCourseDeleteModal';
+
+export const SettingsPanel = () => {
+  const { isOpen: isDeleteModalOpen, onOpen: onDeleteModalOpen, onClose: onDeleteModalClose } = useDisclosure();
```

Comment:
Same as above - why not call them after what they do instead of when they are called.